test/loading/index.js

```diff
@@ -0,0 +1,308 @@
+/* eslint-disable no-new */
+const puppeteer = require('puppeteer-extra');
+const {performance} = require('perf_hooks');
+const scenes = require('../../scenes/scenes.json');
+const axios = require('axios');
+
+const ignoreErrors=  [
+  /** Internal ThreeJS error sometimes gets caught in the extendTexture */
+  // 'at GLTFTextureTransformExtension.extendTexture'
+];
+
+let self;
+
+class LoadTester {
+
+  ignoreList = [
+    'blob'
+  ]
+
+
+  timeOfLastRequest = 0;
+  timeOfSecondLastRequest = 0;
+  lastCheckedAt = 0;
+  lastActivityAt = 0;
+
+  WEBHOOK_URL="https://discord.com/api/webhooks/908078345193930762/m7yVzm8QoTbjhNGJ9NXmOEJTeYSSY1wvX4GlypqLhaxK8MLKnOGiNw7BqXtKRX2z9pu0";
+
+  addStatErr(stat) {
+    if (!this.statsErr) {
+      this.statsErr = [];
+    }
+    var data;
+    var match = /\n/.exec(stat.toString());
+    if (match) {
+      data = `++++ ${this.scene_name} ++++ [ERROR] ++++ File has thrown error \n ${stat} ++++ ${this.scene_name} ++++ [ERROR] ++++ END ERROR`;
+    }
+    else {
+      data = `++++ ${this.scene_name} ++++ [ERROR] ++++ \n ${stat}`;
+    }
+    this.statsErr.push(data);
+  }
+
+  async addStat(type, stat) {
+    if (!this.stats) {
+      this.stats = {
+        errors: [],
+        network: [],
+        performance: [],
+      };
+    }
+    if (type === 'ERROR') {
+      this.stats.errors.push(stat);
+    } else if (type === 'NETWORK') {
+      this.stats.network.push(stat);
+    } else if (type === 'PERFORMANCE') {
+      this.stats.performance.push(stat);
+    } else {
+      this.stats[type] = stat;
+    }
+  }
+
+  waitForServerBoot = ()=>{
+    let retries = 5;
+    const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
+
+    return new Promise((resolve,reject)=>{
+
+      let outInterval = setInterval(async () => {
+        try{
+
+
+              if(retries < 0){
+                clearInterval(outInterval);
+                return reject();  
+              }
+
+              const res = await fetch(self.config.host);
+              if(res.ok){
+                console.log('Webaverse is running tests. Sit tight! :)');
+                clearInterval(outInterval);
+                return resolve();  
+              }
+
+          
+        }catch(e){
+          console.warn(e);
+
+        }finally{
+          retries--;
+        }
+      }, 1000);
+    });
+  }
+  
+  waitForNetworkIdle = ()=>{
```

Comment:
I know this isn't used but I just want to say this will not work for scenes testing. Apps in scenes can totally succeed and still keep the network busy. E.g. loading an infinite list.